### ISSUE TITLE
Eclipse Splash screen image is upside-down on macOS Sonoma (2023-06 and 2023-09 M1)

### ISSUE THREAD

**Issue by minduch** [eclipse-platform/eclipse.platform.swt#751]
This issue is perhaps not in the correct place, it should perhaps be in eclipse.swt or eclipse.pde?

On macOS Sonoma Beta versions, Eclipse 2023-06 and 2023-09 M1 shows the correct image of the splash screen (during workspace selection or just *very* shortly if workspace selection is bypassed), until the progress bar appears, then the image (not the progress bar) is drawn upside-down! I have not seen this on any previous macOS versions such as Ventura.

This also happens to products that are Eclipse PDE based, such as ours with our own splash screen image.

**Comment by Phillipus** [comment]
Is this still happening on the latest Sonoma beta? Can you create a test snippet, perhaps one that loads an image and sets the background image like this:

https://github.com/eclipse-platform/eclipse.platform.ui/blob/fe8052397314a72bfe896e1630fd4882d9ac3482/bundles/org.eclipse.ui.workbench/Eclipse%20UI/org/eclipse/ui/splash/BasicSplashHandler.java#L153-L154

**Comment by kysmith-csg** [comment]
This is happening in the released version as well.

![image (1)](https://github.com/eclipse-platform/eclipse.platform.swt/assets/136501322/dbec310d-f54c-4c4d-b043-1b057e0586eb)

macOS Sonoma 14.0
16-inch 2021 MacBook Pro
Apple M1 Max

**Comment by bearwithtea** [comment]
Happened in mine too. Thought is was some kind of easter egg. Doesn't affect anything besides looks AFAIK.

**Comment by Phillipus** [comment]
The root cause of this has been reported here - https://github.com/eclipse-platform/eclipse.platform.swt/issues/772

**Comment by VineetKanwal1** [comment]
After upgrading my mc to somona, I am seeing first this inverted image on startup and then my eclipse also behaves strage like the tab headers (console, serversm etc) come and go and screen blinks, it has become very hard to work on eclipse after the update for me.

**Comment by travkin79** [comment]
That also happens with the older Eclipse IDE version 2022-09 on MacOS Sonoma 14.0. Surprisingly, the text "IDE" is not shown upside down.

![image](https://github.com/eclipse-platform/eclipse.platform.swt/assets/10887297/05e7d6c3-d6bc-43c6-bc24-bfd72b327c4b)

**Comment by Phillipus** [comment]
> That also happens with the older Eclipse IDE version 2022-09 on MacOS Sonoma 14.0. Surprisingly, the text "IDE" is not shown upside down.

As mentioned, this is because the problem is in macOS Sonoma, not Eclipse. The text is not part of the background image. See #772

**Comment by elsazac** [comment]
```
SonomaSplashScreen [Eclipse Application]	
	org.eclipse.equinox.launcher.Main at localhost:51567	
		Thread [main] (Suspended)	
			Shell(Control).setBackgroundImage(NSImage) line: 3658	
			Shell(Control).updateBackgroundImage() line: 5234	
			Shell(Composite).updateBackgroundImage() line: 1269	
			Shell(Control).setBackgroundImage(Image) line: 3653	
			Workbench$3.run() line: 788	
			SafeRunner.run(ISafeRunnable) line: 47	
			Workbench.createSplashWrapper() line: 829	
			Workbench.lambda$3(Display, WorkbenchAdvisor, int[]) line: 613	
			0x0000000800296550.run() line: not available	
			Realm.runWithDefault(Realm, Runnable) line: 342	
			Workbench.createAndRunWorkbench(Display, WorkbenchAdvisor) line: 552	
			PlatformUI.createAndRunWorkbench(Display, WorkbenchAdvisor) line: 173	
			IDEApplication.start(IApplicationContext) line: 152	
			EclipseAppHandle.run(Object) line: 208	
			EclipseAppLauncher.runApplication(Object) line: 136	
			EclipseAppLauncher.start(Object) line: 104	
			EclipseStarter.run(Object) line: 402	
			EclipseStarter.run(String[], Runnable) line: 255	
			DirectMethodHandle$Holder.invokeStatic(Object, Object, Object) line: not available	
			0x00000008000c4800.invoke(Object, Object, Object, Object) line: not available	
			0x00000008000c4c00.invokeExact_MT(Object, Object, Object, Object, Object) line: not available	
			DirectMethodHandleAccessor.invokeImpl(Object, Object[]) line: 155	
			DirectMethodHandleAccessor.invoke(Object, Object[]) line: 103	
			Method.invoke(Object, Object...) line: 580	
			Main.invokeFramework(String[], URL[]) line: 651	
			Main.basicRun(String[]) line: 588	
			Main.run(String[]) line: 1459	
			Main.main(String[]) line: 1432	
		Thread [Active Thread: Equinox Container: a60850ef-b7b9-42ad-8080-207b043a3b91] (Running)	
		Thread [Framework Event Dispatcher: Equinox Container: a60850ef-b7b9-42ad-8080-207b043a3b91] (Running)	
		Thread [Start Level: Equinox Container: a60850ef-b7b9-42ad-8080-207b043a3b91] (Running)	
		Thread [Refresh Thread: Equinox Container: a60850ef-b7b9-42ad-8080-207b043a3b91] (Running)	
		Thread [Bundle File Closer] (Running)	
		Thread [SCR Component Actor] (Running)	
		Thread [EMF Reference Cleaner] (Running)	
		Thread [Worker-JM] (Running)	
		Thread [Gogo shell] (Running)	
		Thread [Worker-0] (Running)	
		Thread [Git Repository Closer] (Running)	
	/Users/elsazacharia/Documents/JavaVersions2/JDK21/jdk-21.0.1+12/Contents/Home/bin/java (30-Nov-2023, 5:09:26 pm) [pid: 57007]	


```

This is the point where splash screen is getting flipped.

**Comment by babyluoyi** [comment]
How to fix it in 4.16?

**Comment by babyluoyi** [comment]
How to fix it in 4.16?
<img width="800" alt="image" src="https://github.com/eclipse-platform/eclipse.platform.swt/assets/23548657/1a980aaf-29e5-43f6-a00d-e86918a65e96">
